src/ReactiveDomain.Policy/Policy.cs

```diff
@@ -0,0 +1,71 @@
+﻿using ReactiveDomain.Util;
+using System;
+using System.Collections.Generic;
+
+namespace ReactiveDomain.Policy
+{
+    public class Policy : IComparable<Policy>, IComparable, IEquatable<Policy>
+    {
+        public readonly string PolicyName;
+        public readonly HashSet<Role> Roles = new HashSet<Role>();
```

Comment:
Since we have `TryGetRole`, should we make this `private`?

Reply:
For the "sudo" capability, parsing the incoming roles on the `UserDetails` and creating a set of 2 or more `UserPolicies` to switch between is the best way to go here.

On the Policy making the list of roles private makes sense as they should not be changed after initialization and there isn't a really a run time use case I can think of to list all possible roles in a policy.